src/client/clientmap.cpp

```diff
@@ -372,9 +372,13 @@ void ClientMap::updateDrawList()
 		}
 
 		// Block meshes are stored in blocks where all coordinates are even (lowest bit set to 0)
-		// We add both the current block and the block with mesh to the set.
-		shortlist.emplace(block_coord.X, block_coord.Y, block_coord.Z);
+		// Add them to the de-dup set.
 		shortlist.emplace(block_coord.X & ~1, block_coord.Y & ~1, block_coord.Z & ~1);
+		// All other blocks we can grab and add to the drawlist right away.
+		if (block && m_drawlist.emplace(block_coord, block).second) {
```

Comment:
We could perhaps save a few cycles by checking each coordinate and only inserting into the drawlist if this is not a mesh-block... But I don't think it's worth it.